compiler/rustc_mir_build/src/build/expr/category.rs

```diff
@@ -62,6 +62,7 @@ impl Category {
             | ExprKind::Box { .. }
             | ExprKind::Cast { .. }
             | ExprKind::Pointer { .. }
+            | ExprKind::AnonConst { .. }
```

Comment:
This should be like `ExprKind::Literal { .. }`, in the `Constant` arm below.
In general, it would be great to place the new `ExprKind` case below the `Lit`(`eral`) one everywhere, because they're very similar in most cases.